external/storm-hdfs/src/main/java/org/apache/storm/hdfs/spout/HDFSSpout.java

```diff
@@ -722,19 +703,10 @@ private FileReader createFileReader(Path file)
    */
   private FileReader createFileReader(Path file, String offset)
           throws IOException {
-    if ( readerType.equalsIgnoreCase(Configs.SEQ) ) {
-      return new SequenceFileReader(this.hdfs, file, conf, offset);
-    }
-    if ( readerType.equalsIgnoreCase(Configs.TEXT) ) {
-      return new TextFileReader(this.hdfs, file, conf, offset);
-    }
-
     try {
-      Class<?> clsType = Class.forName(readerType);
-      Constructor<?> constructor = clsType.getConstructor(FileSystem.class, Path.class, Map.class, String.class);
+      Constructor<?> constructor = readerType.getConstructor(FileSystem.class, Path.class, Map.class, String.class);
```

Comment:
This assumes a ctor with expected number and type of params is available and tries to invoke it via reflection. Adding an init/open with required params in the `FileReader` is much cleaner IMO and avoids having to check and throw exceptions at runtime.